frontend/frontend-static-pages.php

```diff
@@ -114,7 +114,10 @@ public function pll_pre_translation_url( $url, $language, $queried_object_id ) {
 		if ( ! empty( $queried_object_id ) ) {
 			// Page for posts
 			if ( $GLOBALS['wp_query']->is_posts_page && ( $id = $this->model->post->get( $queried_object_id, $language ) ) ) {
-				$url = get_permalink( $id );
+				$_url = get_permalink( $id );
+				if ( ! empty( $_url ) ) {
+					$url = $_url;
+				}
```

Comment:
Given that the `$url` originally passed is an empty string,we can simplify:
```suggestion
				$url = (string) get_permalink( $id );
```
Or maybe we can take the opportunity to rewrite the method properly:
```php
		if ( empty( $queried_object_id ) ) {
			return $url;
		}

		$id = $this->model->post->get( $queried_object_id, $language );

		if ( empty( $id ) ) {
			return $url;
		}
		
		// Page for posts
		if ( $GLOBALS['wp_query']->is_posts_page ) {
			return (string) get_permalink( $id );
		}

		// Page on front
		if ( is_front_page() && $language->page_on_front === $id ) {
			return $language->get_home_url();
		}

		return $url;
```

Reply:
Done in [c647628](https://github.com/polylang/polylang/pull/1207/commits/c647628b4fd4c5491dbf6d06b458c7e9193608ff).